macros/edr/data_monitoring/monitors_query/table_monitoring_query.sql

```diff
@@ -152,39 +151,46 @@
     {% endif %}
 
     with
-    {%- for metric_name, metric_query in included_monitors.items() %}
+    {%- for metric_name, metric_query in metric_name_to_query.items() %}
         {{ metric_name }} as (
             {{ metric_query }}
         ){% if not loop.last %},{% endif %}
     {%- endfor %}
 
-    {%- for metric_name in included_monitors %}
+    {%- for metric_name in metric_name_to_query %}
     select * from {{ metric_name }}
     {% if not loop.last %} union all {% endif %}
     {%- endfor %}
 {% endmacro %}
 
-{% macro get_metric_query(metric_name, metric_properties) %}
-    {%- set metrics_macro_mapping = {
-        "row_count": elementary.row_count_metric_query,
-        "freshness": elementary.freshness_metric_query,
-        "event_freshness": elementary.event_freshness_metric_query
-    } %}
+{% macro get_metric_query(metric, metric_properties) %}
+    {% if metric_properties.timestamp_column %}
+        {% set metrics_macro_mapping = {
+            "row_count": elementary.row_count_metric_query,
+            "freshness": elementary.freshness_metric_query,
+            "event_freshness": elementary.event_freshness_metric_query
+        } %}
+    {% else %}
+        {% set metrics_macro_mapping = {
+            "row_count": elementary.get_no_timestamp_volume_query,
+            "event_freshness": elementary.get_no_timestamp_event_freshness_query
```

Comment:
Maybe let's ensure that we have at least a sanity test for each of the "no timestamps" (I'm not 100% sure if we do or not)

Reply:
We do for [volume](https://github.com/elementary-data/dbt-data-reliability/blob/5f2ad803e76dcadf6e5d483c3d24705a015966f3/integration_tests/tests/test_volume_anomalies.py#L195), I'll add one for `event_freshness`.